ADR/partner-tasks.md

```diff
@@ -0,0 +1,34 @@
+# Partner Tasks in Build/Test Pipelines
+
+Date: 2023-03-06
+
+## Status
+
+Provisional
+
+## Context
+
+* As a Red Hat Partner, I would like to offer our service's capability as a Tekton Task that would be executed in a user's build/test Pipeline on StoneSoup.
```

Comment:
After reviewing the ADR, I realize that I might be conflating "Partner" with "user" or "customer." Can we define what it means to be a partner? Is there any criteria that is met? Is there a process to become one? Is this just for certain tools that want to be leveraged in the pipeline (i.e. `random-sast-scanner`) or might this also be relevant if a user wants to contribute a task that they developed back to the platform?

Reply:
> an we define what it means to be a partner? Is there any criteria that is met? Is there a process to become one? 

This process doesn't quite care about it at the moment. BU should let engineering know that 'someone' wants to contribute their Task into our build/test Pipeline.

Internal or external partner - the stonesoup engineering team is notified that we should be expecting a contribution from a specific github handle. Once we are sure its a legitimate contribution ( and CI has OK'd it ). Also, we shall take a call on whether the image used in the Tekton Task needs to be in registry.redhat.io .